views.go

```diff
@@ -1,14 +1,68 @@
 package main
 
 import (
+	"bytes"
 	"database/sql"
 	"fmt"
 	"github.com/labstack/echo"
+	"io/ioutil"
 	"net/http"
 	"net/url"
 	"strings"
 )
 
+func RegisterGET(c echo.Context) error {
+
+	data := NewTemplateGlobal()
+
+	service := ValidateService(c)
+	if service == nil {
+		data.NotAuthorized = true
+		return c.Render(http.StatusUnauthorized, "register", data)
+	}
+	data.CASRegister = GetCasRegisterUrl(service.String())
+	data.CASLogin = GetCasLoginUrl(service.String())
+	return c.Render(http.StatusOK, "register", data)
+}
+
+func RegisterPOST(c echo.Context) error {
+
+	data := NewTemplateGlobal()
+
+	service := ValidateService(c)
+	if service == nil {
+		data.NotAuthorized = true
+		return c.Render(http.StatusUnauthorized, "register", data)
+	}
+	data.CASRegister = GetCasRegisterUrl(service.String())
+	data.CASLogin = GetCasLoginUrl(service.String())
+
+	fullname := strings.TrimSpace(c.FormValue("fullname"))
+	email := strings.ToLower(strings.TrimSpace(c.FormValue("email")))
+	password := c.FormValue("password")
+
+	api_v1_url_for_register_user := "http://" + *OSFHost + "/api/v1/register/"
+	jsonStr := "{\"email1\": \"" + email + "\", \"email2\": \"" + email + "\", \"password\": \"" + password + "\", \"fullName\": \"" + fullname + "\"}"
+	byteStr := []byte(jsonStr)
+	resp, err := http.Post(api_v1_url_for_register_user, "application/json", bytes.NewBuffer(byteStr))
+	if err != nil {
+		panic(err)
```

Comment:
`panic` will stop execution

Reply:
`panic(err)` is used for four other places in the `views.go`. For example, https://github.com/CenterForOpenScience/fakecas/blob/develop/views.go#L34. Should these be updated as well?

Reply:
Those are for specifically unrecoverable errors and have if guards.